twarep/electrifly
# Insights Page with Activity Breakdown

Note that this is a PR for 2 tickets: "Update the stats section with the labelled data" and "Statistical Insights: Weather"
## Checklist
**Note:** Put an x into the box like this [x] to check these off

## What I Did
In this PR, there are 2 graphics. The first graphic is a scatter plot with power vs. SOC rate of change, and it is coloured by activity with lines of best fit for each activity. The second graphic is a table where it breaks down the SOC rate of change for each activity.

app.py: to output the plots and table
flight_querying.py: added queries to extract data from views
Graphing.py: to set up the scatter plots for output in app.py

## How I Did it?
Used matplotlib for the scatter plots.

## How to Test?
Go to Data Analysis -> Insights, then manually test. Try selecting different date filters and activity filters. Note that it takes a while to load the plots, make sure only change 1 filter at a time.

### Relevant Links
Any tutorials or other resources used to help with this task. You can also cite the documentation.
